### ISSUE TITLE
Woocommerce 4.2.1 update, linked products, cannot delete, X is missing

### ISSUE THREAD

**Issue by reason8910** [woocommerce/woocommerce#26845]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
Woocommerce 4.2.1 update, linked products, cannot delete as the "x" is now missing from the upsell and cross-sell products. 
Tested this before and after 2 separate installations.
Also cleared cache to make sure.

**Expected behavior**
In version 4.2.0 there was an x on each product n the upsell field so you could remove it. 

**Actual behavior**
Now its gone.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. View upsells pre update to version 4.2.1 
2. See x on upsell products in the field
3. Update to 4.2.1
4. Check upsells and "x" is now missing
![upsells](https://user-images.githubusercontent.com/5126311/85363271-ce8da300-b563-11ea-8e84-3f598813ea6b.jpg)

**Screenshots**

**Isolating the problem (mark completed items with an [x]):**
- [x ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
The System Status Report is found in your WordPress admin under **WooCommerce > Status**. 
Please select “Get system report”, then “Copy for support”, and then paste it here.
`
### WordPress Environment ###

WordPress address (URL): 
Site address (URL): 
WC Version: 4.2.1
REST API Version: ✔ 1.0.8
WC Blocks Version: ✔ 2.5.16
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ❌ Inactive 1.2.4
Log Directory Writable: ✔
WP Version: 5.4.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.2.31
PHP Post Max Size: 40 MB
PHP Time Limit: 14400
PHP Max Input Vars: 1000
cURL Version: 7.62.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.48-cll-lve
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.2.1
WC Database Prefix: 
Total Database Size: 173.10MB
Database Data Size: 132.09MB
Database Index Size: 41.01MB
wxp4_woocommerce_sessions: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wxp4_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wxp4_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_woocommerce_order_itemmeta: Data: 0.17MB + Index: 0.14MB + Engine InnoDB
wxp4_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wxp4_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_actionscheduler_actions: Data: 23.56MB + Index: 16.61MB + Engine InnoDB
wxp4_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_actionscheduler_logs: Data: 18.55MB + Index: 10.03MB + Engine InnoDB
wxp4_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_comments: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wxp4_ebay_accounts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_auctions: Data: 1.27MB + Index: 0.14MB + Engine InnoDB
wxp4_ebay_categories: Data: 1.52MB + Index: 0.56MB + Engine InnoDB
wxp4_ebay_jobs: Data: 0.16MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_log: Data: 0.17MB + Index: 0.05MB + Engine InnoDB
wxp4_ebay_messages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_orders: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wxp4_ebay_payment: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_profiles: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_shipping: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_sites: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_stocks_log: Data: 17.55MB + Index: 0.00MB + Engine InnoDB
wxp4_ebay_store_categories: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_ebay_transactions: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wxp4_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_mobileassistant_accounts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_mobileassistant_devices: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_mobileassistant_failed_login: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_mobileassistant_push_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_mobileassistant_session_keys: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_mobileassistant_users: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_odb_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_options: Data: 2.45MB + Index: 0.22MB + Engine InnoDB
wxp4_pmxi_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_pmxi_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_pmxi_images: Data: 0.09MB + Index: 0.00MB + Engine InnoDB
wxp4_pmxi_imports: Data: 0.11MB + Index: 0.00MB + Engine InnoDB
wxp4_pmxi_posts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_pmxi_templates: Data: 0.14MB + Index: 0.00MB + Engine InnoDB
wxp4_postmeta: Data: 43.58MB + Index: 7.03MB + Engine InnoDB
wxp4_posts: Data: 8.52MB + Index: 3.75MB + Engine InnoDB
wxp4_termmeta: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wxp4_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_term_relationships: Data: 0.23MB + Index: 0.14MB + Engine InnoDB
wxp4_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_usermeta: Data: 3.52MB + Index: 0.27MB + Engine InnoDB
wxp4_users: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wxp4_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_wc_gpf_render_cache: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wxp4_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wxp4_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wxp4_wc_product_meta_lookup: Data: 0.28MB + Index: 0.44MB + Engine InnoDB
wxp4_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wxp4_wfconfig: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wxp4_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wffilemods: Data: 3.52MB + Index: 0.00MB + Engine InnoDB
wxp4_wfhits: Data: 0.09MB + Index: 0.05MB + Engine InnoDB
wxp4_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wxp4_wfknownfilelist: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wxp4_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wflogins: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wxp4_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wxp4_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wxp4_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wxp4_wfstatus: Data: 0.13MB + Index: 0.11MB + Engine InnoDB
wxp4_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wxp4_woocommerce_gpf_google_taxonomy: Data: 1.52MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 12923
customize_changeset: 14
fl-builder-template: 1
nav_menu_item: 24
page: 13
post: 2
product: 2024
product_variation: 10
revision: 1
shop_coupon: 3
shop_order: 69
shop_order_refund: 2
shop_quote: 10
wc_custom_statuses: 8

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (35) ###

AddToAny Share Buttons: by AddToAny – 1.7.41
Akismet Anti-Spam: by Automattic – 4.1.6
Autoremove Attachments: by Polygon Themes – 1.1.2
Category Banner Management for Woocommerce: by theDotstore – 2.0.4 – Installed version not tested with active version of WooCommerce 4.2.1
Beaver Builder Plugin (Pro Version): by The Beaver Builder Team – 2.3.2.5
Classic Editor: by WordPress Contributors – 1.5
Customizer Export/Import: by The Beaver Builder Team – 0.9.1
Disable WooCommerce Bloat: by ospiotr – 1.1 – Installed version not tested with active version of WooCommerce 4.2.1
Easy WP SMTP: by wpecommerce
alexanderfoxc – 1.3.9.3

GAinWP Google Analytics Integration for WordPress: by IntelligenceWP – 5.4.5
Google XML Sitemaps: by Arne Brachhold – 4.1.0
LiteSpeed Cache: by LiteSpeed Technologies – 3.2.3.2
Mailchimp for WooCommerce: by Mailchimp – 2.4.1 – Installed version not tested with active version of WooCommerce 4.2.1
Mobile Assistant Connector: by eMagicOne – 2.1.5
Reason8 - Message after Product Details: by Reason8 – 1.0
Reason8 - Product BIN Location in Warehouse: by Reason8 – 1.0 – Installed version not tested with active version of WooCommerce 4.2.1
Reason8 - RSS Feed with Image and Price: by Reason8 – 1.0
Woocommerce Shipping Calculator On Product Page |  VestaThemes.com: by Magerips – 1.8 – Installed version not tested with active version of WooCommerce 4.2.1
Optimize Database after Deleting Revisions: by CAGE Web Design | Rolf van Gelder
Eindhoven
The Netherlands – 4.9.1

Search By Product SKU - for Woocommerce: by Rohit kumar – 1.0.0 – Installed version not tested with active version of WooCommerce 4.2.1
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.16.25

WooCommerce Auto Category Thumbnails: by Shellbot – 1.2.1 – Installed version not tested with active version of WooCommerce 4.2.1
WooCommerce Conversion Tracking: by weDevs – 2.0.8 – Installed version not tested with active version of WooCommerce 4.2.1
WooCommerce Cost of Goods: by SkyVerge – 2.9.8
Print Invoice & Delivery Notes for WooCommerce: by Tyche Softwares – 4.5.5 – Installed version not tested with active version of WooCommerce 4.2.1
WooCommerce Google Analytics Integration: by WooCommerce – 1.4.22
WooCommerce Menu Cart: by Jeremiah Prummer
Ewout Fernhout – 2.9.2

WooCommerce Google Product Feed: by Ademti Software Ltd. – 8.1.2
WooCommerce Protected Categories: by Barn2 Media – 2.3.2 – Installed version not tested with active version of WooCommerce 4.2.1
WooCommerce Order Status & Actions Manager: by Actuality Extensions – 2.4.11 – Installed version not tested with active version of WooCommerce 4.2.1
WooCommerce Waitlist: by Neil Pie – 2.1.20
WooCommerce: by Automattic – 4.2.1
Wordfence Security: by Wordfence – 7.4.8
WP-Lister Pro for eBay: by WP Lab – 2.5.4 – Installed version not tested with active version of WooCommerce 4.2.1
Zip - WooCommerce: by Zip – 2.2.1 – Installed version not tested with active version of WooCommerce 4.2.1

### Inactive Plugins (1) ###

WPD Beaver Popups: by Doug Belchamber – 1.1.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude_from_catalog (exclude_from_catalog)
exclude_from_search (exclude_from_search)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Beaver Builder Child Theme
Version: 1.0
Author URL: http://www.fastlinemedia.com
Child Theme: ✔
Parent Theme Name: Beaver Builder Theme
Parent Theme Version: 1.7.6.3
Parent Theme Author URL: http://www.wpbeaverbuilder.com/?utm_medium=bb-pro&utm_source=bb-theme&utm_campaign=themes-admin-page
WooCommerce Support: ✔

### Templates ###

Overrides: –

### WooCommerce Google Product Feed fields ###

Product description: Main product description (full preferred) plus variation description.
Availability: Defaults to "in stock".
Condition: Defaults to "new".
Brand: Defaults to "YDS.com.au".
Manufacturer Part Number (MPN): Pre-populates from product sku.
Product Type: Defaults to "Home & Garden". Pre-populates from product_cat taxonomy.
Google Product Category: Defaults to "Home & Garden".
Global Trade Item Number (GTIN): 
Size system: Defaults to "AU".
Adult content: Defaults to "no".
Identifier exists flag: Defaults to "not-included".
Adwords labels: Pre-populates from product_cat taxonomy.

### WooCommerce Google Product Feed options ###

Include variations in feed: No
Send item_group_id: No
debug_key: c2c56b1d-663c-419d-93e6-69b07d65935a

### Action Scheduler ###

Complete: 6,094
Oldest: 2020-05-23 15:21:34 +1000
Newest: 2020-06-23 15:01:44 +1000

Failed: 1
Oldest: 2020-02-21 10:08:34 +1100
Newest: 2020-02-21 10:08:34 +1100

Pending: 2
Oldest: 2020-06-23 15:16:44 +1000
Newest: 2020-06-23 15:16:45 +1000

`
</details>

**Comment by pedropmv** [comment]
Same here

**Comment by tammullen** [comment]
Hi @reason8910 

Thank you for taking the time reporting this bug, we really appreciate your help. I can reproduce it on my end. 

We’re going to solve this bug as soon as possible. Stay tuned for updates.

**Comment by peterfabian** [comment]
Included in both 4.3-rc.1 and 4.2.2 now.

**Comment by reason8910** [comment]
Thanks for the quick fix.
